Lib/dataclasses.py

```diff
@@ -0,0 +1,776 @@
+
+
+def _set_attribute(cls, name, value):
+    # Raise TypeError if an attribute by this name already exists.
+    if name in cls.__dict__:
+        raise TypeError(f'Cannot overwrite attribute {name} '
+                        f'in {cls.__name__}')
+    setattr(cls, name, value)
+
+
+def _process_class(cls, repr, eq, order, hash, init, frozen):
+    # Use an OrderedDict because:
+    #  - Order matters!
+    #  - Derived class fields overwrite base class fields, but the
+    #    order is defined by the base class, which is found first.
+    fields = collections.OrderedDict()
+
+    # Find our base classes in reverse MRO order, and exclude
+    #  ourselves.  In reversed order so that more derived classes
+    #  override earlier field definitions in base classes.
+    for b in cls.__mro__[-1:0:-1]:
+        # Only process classes that have been processed by our
+        #  decorator.  That is, they have a _MARKER attribute.
+        base_fields = getattr(b, _MARKER, None)
+        if base_fields:
+            for f in base_fields.values():
+                fields[f.name] = f
+
+    # Now find fields in our class.  While doing so, validate some
+    #  things, and set the default values (as class attributes)
+    #  where we can.
+    for f in _find_fields(cls):
+        fields[f.name] = f
+
+        # If the class attribute (which is the default value for
+        #  this field) exists and is of type 'Field', replace it
+        #  with the real default.  This is so that normal class
+        #  introspection sees a real default value, not a Field.
+        if isinstance(getattr(cls, f.name, None), Field):
+            if f.default is _MISSING:
+                # If there's no default, delete the class attribute.
+                #  This happens if we specify field(repr=False), for
+                #  example (that is, we specified a field object, but
+                #  no default value).  Also if we're using a default
+                #  factory.  The class attribute should not be set at
+                #  all in the post-processed class.
+                delattr(cls, f.name)
+            else:
+                setattr(cls, f.name, f.default)
+
+    # Remember all of the fields on our class (including bases).  This
+    #  marks this class as being a dataclass.
+    setattr(cls, _MARKER, fields)
+
+    # We also need to check if a parent class is frozen: frozen has to
+    #  be inherited down.
+    is_frozen = frozen or cls.__setattr__ is _frozen_setattr
+
+    # If we're generating ordering methods, we must be generating
+    #  the eq methods.
+    if order and not eq:
+        raise ValueError('eq must be true if order is true')
+
+    if init:
+        # Does this class have a post-init function?
+        has_post_init = hasattr(cls, _POST_INIT_NAME)
+
+        # Include InitVars and regular fields (so, not ClassVars).
+        _set_attribute(cls, '__init__',
+                       _init_fn(list(filter(lambda f: f._field_type
```

Comment:
Could you please extract the lambda as a local function? This code is too complex.

Reply:
I'll consider it. These expressions started out much simpler, and grew over time.